crates/shared/src/http_solver.rs

```diff
@@ -173,6 +177,37 @@ impl HttpSolverApi for DefaultHttpSolverApi {
         serde_json::from_str(text)
             .with_context(|| format!("failed to decode response json, {}", context()))
     }
+
+    fn notify_auction_result(&self, auction_id: AuctionId, result: model::AuctionResult) {
+        let mut url = match self.base.join("notify") {
+            Ok(url) => url,
+            Err(err) => {
+                tracing::warn!(?err, "failed to create notify url");
+                return;
+            }
+        };
+
+        let client = self.client.clone();
+        let config_api_key = self.config.api_key.clone();
+        let future = async move {
+            url.query_pairs_mut()
+                .append_pair("auction_id", auction_id.to_string().as_str());
+
+            let mut request = client
+                .post(url)
+                .header(header::CONTENT_TYPE, "application/json")
+                .header(header::ACCEPT, "application/json");
+
+            if let Some(api_key) = config_api_key {
+                let mut header = HeaderValue::from_str(api_key.as_str()).unwrap();
+                header.set_sensitive(true);
+                request = request.header("X-API-KEY", header);
+            }
+
+            let _result = request.json(&json!(result)).send().await;
```

Comment:
Should maybe debug log that we are notififying solvers (for debugging purposes later)

Reply:
Indeed. Added.